checker/src/block_visitor.rs

```diff
@@ -1749,6 +1749,10 @@ impl<'block, 'analysis, 'compilation, 'tcx, E>
                                 if let Some(arg_val) = gen_args.as_ref().get(*index as usize) {
                                     return self.visit_constant(None, arg_val.expect_const());
                                 }
+                            } else {
```

Comment:
This can go away if you get rid of Sealed. There is still a problem here, but we need a different test case for it (and and issue, at some point).